pkg/api/handlers/utils/containers.go

```diff
@@ -27,32 +26,92 @@ func WaitContainer(w http.ResponseWriter, r *http.Request) (int32, error) {
 		Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrapf(err, "failed to parse parameters for %s", r.URL.String()))
 		return 0, err
 	}
-	if _, found := r.URL.Query()["interval"]; found {
-		interval, err = time.ParseDuration(query.Interval)
-		if err != nil {
-			InternalServerError(w, err)
-			return 0, err
-		}
-	} else {
-		interval, err = time.ParseDuration("250ms")
-		if err != nil {
-			InternalServerError(w, err)
-			return 0, err
-		}
-	}
-	condition := define.ContainerStateStopped
+
+	var condition string
 	if _, found := r.URL.Query()["condition"]; found {
-		condition, err = define.StringToContainerStatus(query.Condition)
-		if err != nil {
-			InternalServerError(w, err)
+		condition = query.Condition
+		if condition != "next-exit" && condition != "removed" && condition != "not-running" && condition != "" {
```

Comment:
I think the Libpod version of this has to continue accepting the old conditions as well to ensure backwards compat for that API.